src/connectors/hierarchical-menu/connectHierarchicalMenu.js

```diff
@@ -197,9 +197,17 @@ export default function connectHierarchicalMenu(renderFn, unmountFn) {
         unmountFn();
 
         // compute nextState for the search
-        const nextState = state
-          .removeHierarchicalFacetRefinement(hierarchicalFacetName)
-          .removeHierarchicalFacet(hierarchicalFacetName);
+        let nextState;
+
+        if (state.isHierarchicalFacetRefined(hierarchicalFacetName)) {
+          nextState = state.removeHierarchicalFacetRefinement(
+            hierarchicalFacetName
+          );
+        }
+
+        nextState = (nextState || state).removeHierarchicalFacet(
```

Comment:
If you initialize `nextState` with `state` at l.200 you don't need to use `nextState || state`.